upload/catalog/controller/smailyforopencart/order.php

```diff
@@ -17,4 +17,23 @@ public function removeSent() {
             );
         }
     }
+
+    /**
+     * Removes customer from abandoned carts table when clearing all items from cart.
+     * Event handler for catalog/controller/checkout/cart/remove/after.
+     *
+     * @return void
+     */
+    public function removeWhenCartEmpty() {
+        $customer_id = $this->customer->getId();
+        if (isset($customer_id)) {
```

Comment:
Maybe it would be better to check here if `$customer_id` is integer? `true` can pass this if-condition and `(int) true === 1`.

Reply:
`$this->customer->getId();` will return `NULL` when not logged-in customer manages cart.So  `isset — Determine if a variable is declared and is different than NULL` seems OK to me. The reason that I didn't want to use `(int) $this->customer->getId();` is because `(int) NULL` is `0` and there might be account with id `0`.

Also the `$this->customer->getId()` will return a string representation of the id number. Should be `int` in my opinion but it is as it is.

Reply:
OK. If `NULL` is guaranteed, then let's leave it as-is.